libstuff/libstuff.cpp

```diff
@@ -525,7 +525,6 @@ bool SParseList(const char* ptr, list<string>& valueList, char separator) {
 
 // --------------------------------------------------------------------------
 void SConsumeFront(string& lhs, ssize_t num) {
-    SASSERT((int)lhs.size() >= num);
```

Comment:
Hm, why remove this?  The only time this alerts is if we think we sent *more* than the buffer size, which seems pretty extreme.

Reply:
Yeah... Tyler and I are discussing in slack, we're not really sure how we died here now, as sending more than what's in the buffer shouldn't be possible.

Reply:
Ok, then let's not remove this as it sounds like we don't actually understand the real bug.  This is a protection; removing the protection isn't a great solution.

Reply:
Agreed, we're looking for a better solution.